src/utils/OnboardingProvider.tsx

```diff
@@ -76,11 +79,12 @@ export default function OnboardingProvider({
     };
 
     const flushData = async () => {
+      if (!auth) throw new Error('Fatal: No auth context provided!');
       if (!profileCtx) throw new Error('Fatal: No profile context provided!');
-      if (profileCtx.userId === undefined)
+      if (auth.userId === undefined)
```

Comment:
i would say it's better to use `!auth.userId` here instead (idk what i was thinking before) since it would check for all of:
1. `''`
2. `null`
3. `undefined`